tracker/src/main/java/org/matomo/sdk/Tracker.java

```diff
@@ -192,6 +192,15 @@ public void setDispatchTimeout(int timeout) {
         mDispatcher.setConnectionTimeOut(timeout);
     }
 
+    /**
+     * For the Dispatcher to run in offline mode.  This will send all tracked events to disk until the next time the tracker is constructed.
+     *
+     * @see Dispatcher#setOffline()
+     */
+    public void setOffline() {
+        mDispatcher.setOffline();
```

Comment:
Good point on the mode restoration via preferences. Have not thought about that, but I'm not totally against just making this an edge cases, e.g. call it `setDispatchMode(Mode.EXCEPTION)` and then don't persist this setting.

I'll think about it a bit more.

Reply:
Sounds good.  I went back and forth on it a lot.

Reply:
I would favor the approach of introducing a `Mode.EXCEPTION` that allows for special handling. Future changes can expand on this with different modes and it seems like it would create easier to understand code. e.g. We are in Mode X, so we do Y, instead of "Flag A, B and C are set, let's do D". So a `Mode.EXCEPTION` with a check that it's not persisted to preferences (need a test for that).

What are your thoughts?

Reply:
Makes sense to me.  `DispatchMode` seemed like the logical place to put it but I didn't want to vary from its original meaning without understanding more about what was going on there.  If that works for you I'll write it up (and test)

Reply:
Hm, while we persist the mode in the tracker, the mode isn't actually restored to the dispatcher afaict. So on re-init the dispatcher uses the default mode again. Definitely a bug, the question is whether to fix it which would change the current behavior, or remove the functionality which would bring in it in line with the current behavior.

Reply:
It's now correctly restored.